src/App/Controller/ProposalController.php

```diff
@@ -84,8 +86,48 @@ public function editAction(Request $request, $slug)
              ]));
         }
 
-        return $this->render('App:Proposal:edit.html.twig', [
-            'form' => $form->createView(),
+        return $this->render('App:Proposal:edit_proposal.html.twig', [
+            'proposal' => $proposal,
+            'form'     => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_USER')")
+     */
+    public function editMotivationAction(Request $request, $slug)
+    {
+        $proposal = $this->getDoctrine()->getRepository('App:Proposal')->findOneBySlug($slug);
+
+        if (null ===  $proposal) {
+            throw $this->createNotFoundException();
+        }
+
+        if (false === $this->get('security.context')->isGranted('edit', $proposal)) {
+            throw new AccessDeniedException();
+        }
+
+        $oldProposal = new OldProposal($proposal);
+        $form        = $this->createForm(new EditMotivationProposalType(), $proposal);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $proposal
+                ->addToHistory($oldProposal)
+                ->setVersionNumber($proposal->getVersionNumber() + 1)
```

Comment:
You should have a method `->incrementVersionNumber()`.

Reply:
Done